silo/src/config.re

```diff
@@ -0,0 +1,123 @@
+open Toml;
+
+type config_load_error =
+  | ConfigNotFound
+  | ConfigParseError(string, Parser.location)
+  | ConfigMissingRequiredField(string);
+
+exception LoadError(config_load_error);
+
+type t = {
+  package: package_config,
+  build: build_config,
+  bin: bin_config,
+}
+and package_config = {name: string}
+and build_config = {
+  elide_type_info: option(bool),
+  include_dirs: option(list(string)),
+  import_memory: option(bool),
+  initial_memory_pages: option(int),
+  maximum_memory_pages: option(int),
+  memory_base: option(int),
+  gc: option(bool),
+  stdlib: option(string),
+  use_start_section: option(bool),
+  wasi_polyfill: option(string),
+  wasm_features,
+}
+and wasm_features = {
+  bulk_memory: option(bool),
+  tail_call: option(bool),
+}
+and bin_config = {
+  name: option(string),
+  path: option(string),
+  wat: option(bool),
+  source_map: option(bool),
+};
+
+let load = () => {
+  switch (Parser.from_filename("silo.toml")) {
+  | `Ok(config) =>
+    open Lenses;
+
+    let package =
+      switch (get(config, field("package"))) {
+      | Some(package) => package
+      | None => raise(LoadError(ConfigMissingRequiredField("[package]")))
+      };
+
+    let package_config = {
+      name:
+        switch (get(package, key("name") |-- string)) {
+        | Some(name) => name
+        | None =>
+          raise(LoadError(ConfigMissingRequiredField("name of [package]")))
+        },
+    };
+
+    let build = field("build");
+
+    let wasm_features = build |-- field("wasm-features");
+    let wasm_features = {
+      bulk_memory:
+        get(config, wasm_features |-- key("bulk-memory") |-- bool),
+      tail_call: get(config, wasm_features |-- key("tail-call") |-- bool),
+    };
+
+    let build_config = {
+      elide_type_info:
+        get(config, build |-- key("elide-type-info") |-- bool),
```

Comment:
I'm curious about the choice of kebab case? The convention within the language itself is camel case so why not use that here?

Afaik there's no key naming convention in TOML and I've seen a mixture of all of
* Rust, Gleam: snake case
* Python: kebab case (odd that it's not snake case...)
* Hugo: camel case

Reply:
Rust uses kebab case, so that + mostly translating command line flags is what got me here. I'd be open to camel case though. I sort of like the configuration language being different than the source language though.

Reply:
Ah yeah rust does use kebab case, my bad. Same question about package names and output binary name conventions: I see in the PR description you have kebab case but in the stdlib the convention so far has been camel case. When package management, etc rolls out are we going with kebab case package names like npm?